js/goto.js

```diff
@@ -1,27 +1,42 @@
+import {
+  SIGNUP,
+  JOIN,
+  HOME,
+  SIGNUP_URL,
+  JOIN_URL,
+  HOME_URL,
+} from './constants.js';
+
+const hasJoinVisited = localStorage.getItem('hasJoinVisited');
+
 function redirectUserToPage(page) {
   const finalPage =
-    page == 'signup'
-      ? 'https://my.realdevsquad.com/signup?state=get-started'
-      : 'https://realdevsquad.com';
+    page == SIGNUP ? SIGNUP_URL : page == JOIN ? JOIN_URL : HOME_URL;
 
   window.location.href = finalPage;
 }
 
+function redirectionHandler(data) {
```

Comment:
`[Alternative]Suggestion`
```js
function redirectionHandler(data) {
  const hasVisitedJoin = hasJoinVisited == 'true' || hasJoinVisited == null;
  
  const nextPage = data?.incompleteUserDetails ? SIGNUP : hasVisitedJoin ? JOIN : HOME;
  
  redirectUserToPage(nextPage);
}
```

Reply:
```js
const PAGE_MAPPING = [
  {
    condition: data.incompleteUserDetails,
    page: SIGNUP,
  },
  {
    condition: hasJoinVisited === 'true' || hasJoinVisited === null,
    page: JOIN,
  },
];

function redirectionHandler(data) {
  let nextPage = HOME; // default page

  for (const { condition, page } of PAGE_MAPPING) {
    if (condition(data)) {
      nextPage = page;
      break;
    }
  }
  
  redirectUserToPage(nextPage);
}
```

Another Alternative!!

Reply:
this changes is necessary can we can't keep it simple ?